packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -43,18 +44,34 @@ export interface IOutboxParameters {
 	readonly logger: ITelemetryLoggerExt;
 	readonly groupingManager: OpGroupingManager;
 	readonly getCurrentSequenceNumbers: () => BatchSequenceNumbers;
+	readonly reSubmit: (
+		content: string,
+		localOpMetadata: unknown,
+		opMetadata: Record<string, unknown> | undefined,
+	) => void;
+	readonly opReentrancy: () => boolean;
+	readonly closeContainer: (error?: ICriticalContainerError) => void;
 }
 
-function getLongStack(action: () => Error): Error {
-	// Increase the stack trace limit temporarily, so as to debug better in case it occurs.
+/**
+ * Temporarily increase the stack limit while executing the provided action.
+ * If a negative value is provided for `length`, no stack frames will be collected.
+ * If Infinity is provided, all frames will be collected.
+ *
+ * ADO:4663 - add this to the common packages.
+ *
+ * @param action - action which returns an error
+ * @param length - number of stack frames to collect, 50 if unspecified.
+ * @returns the result of the action provided
+ */
+export function getLongStack<T>(action: () => T, length: number = 50): T {
+	const originalStackTraceLimit = (Error as any).stackTraceLimit;
+
 	try {
-		const originalStackTraceLimit = (Error as any).stackTraceLimit;
-		(Error as any).stackTraceLimit = 50;
-		const result = action();
-		(Error as any).stackTraceLimit = originalStackTraceLimit;
-		return result;
-	} catch (error) {
+		(Error as any).stackTraceLimit = length;
 		return action();
+	} finally {
+		(Error as any).stackTraceLimit = originalStackTraceLimit;
```

Comment:
Is it guaranteed that this property will exist for assigning? (I have a feeling this is something I ran into and why I wrapped everything, or I could've just been over-cautious)

Reply:
hmm... interesting. I look deeper into this. Thanks!

Reply:
The issue is that if `action` throws, we'd never revert the stack trace limit. I think we can try to figure out if the object really does have a writeable property..